adoc/bp_chap_getting_started_with_salt_formulas.adoc

```diff
@@ -331,13 +331,43 @@ Many of these values are self-explanatory:
 The ``group``, ``edit-group``, and `namespace` (formerly ``hidden-group``) types do not generate an editable field and are used to structure form and pillar data.
 The difference between `group` and `namespace` is `group` generates a visible border with a heading, and `namespace` shows nothing visually (and is only used to structure pillar data).
 The difference between `group` and `edit-group` is: `edit-group` allows to structure and restrict editable fields in a more flexible way.
+`edit-group` is a collection of items of the same kind; collections can have the following four "shapes":
+
+** A list of primitive items
+** A list of dictionaries
+** A dictionary of primitive items
+** A dictionary of dictionaries
+
+The size of such collection is variable; users can add or remove elements.
+
 For example, `edit-group` supports the `$minItems` and `$maxItems` attributes, and thus it simplifies complex and repeatable input structures.
+These, and also `itemName`, are optional.
 For an `edit-group` example, see <<salt.formulas.ex.edit-group>>.
 
 $default::
 `$default` allows you to specify a default value that is displayed and used, if no other value is entered.
 In an `edit-group` it allows to create initial members of the group and populate them with specified data.
 
+$optional::
+`$optional` is a boolean attribute.
+If it is `true` and the field is empty in the form, then this field will not be generated in the formula data.
+The generated dictionary will miss the field name key.
```

Comment:
Is there a more specific word than 'miss'? Possibly 'skip'?

Reply:
I tried a different version. @hustodemon can you please check whether it makes sense?